File: windows/Liferay.SDK/Service/V72/Kbfolder/KbfolderService.cs
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Liferay.SDK.Service.V72.Kbfolder
{
	public class KbfolderService : ServiceBase
	{
		public KbfolderService(ISession session)
			: base(session)
		{
		}

		public async Task<dynamic> DeleteKbFolderAsync(long kbFolderId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("kbFolderId", kbFolderId);

			var _command = new JsonObject()
			{
				{ "/kb.kbfolder/delete-kb-folder", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetKbFoldersAsync(long groupId, long parentKBFolderId, int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("parentKBFolderId", parentKBFolderId);
			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/kb.kbfolder/get-kb-folders", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<long> GetKbFoldersCountAsync(long groupId, long parentKBFolderId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("parentKBFolderId", parentKBFolderId);

			var _command = new JsonObject()
			{
				{ "/kb.kbfolder/get-kb-folders-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<dynamic> AddKbFolderAsync(long groupId, long parentResourceClassNameId, long parentResourcePrimKey, string name, string description, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("parentResourceClassNameId", parentResourceClassNameId);
			_parameters.Add("parentResourcePrimKey", parentResourcePrimKey);
			_parameters.Add("name", name);
			_parameters.Add("description", description);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/kb.kbfolder/add-kb-folder", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> UpdateKbFolderAsync(long parentResourceClassNameId, long parentResourcePrimKey, long kbFolderId, string name, string description)
		{
			var _parameters = new JsonObject();

			_parameters.Add("parentResourceClassNameId", parentResourceClassNameId);
			_parameters.Add("parentResourcePrimKey", parentResourcePrimKey);
			_parameters.Add("kbFolderId", kbFolderId);
			_parameters.Add("name", name);
			_parameters.Add("description", description);

			var _command = new JsonObject()
			{
				{ "/kb.kbfolder/update-kb-folder", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> UpdateKbFolderAsync(long parentResourceClassNameId, long parentResourcePrimKey, long kbFolderId, string name, string description, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("parentResourceClassNameId", parentResourceClassNameId);
			_parameters.Add("parentResourcePrimKey", parentResourcePrimKey);
			_parameters.Add("kbFolderId", kbFolderId);
			_parameters.Add("name", name);
			_parameters.Add("description", description);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/kb.kbfolder/update-kb-folder", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> FetchKbFolderAsync(long kbFolderId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("kbFolderId", kbFolderId);

			var _command = new JsonObject()
			{
				{ "/kb.kbfolder/fetch-kb-folder", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> FetchFirstChildKbFolderAsync(long groupId, long kbFolderId, JsonObjectWrapper obc)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("kbFolderId", kbFolderId);
			this.MangleWrapper(_parameters, "obc", "$languageUtil.getJSONWrapperClassName($parameter.type)", obc);

			var _command = new JsonObject()
			{
				{ "/kb.kbfolder/fetch-first-child-kb-folder", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> FetchFirstChildKbFolderAsync(long groupId, long kbFolderId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("kbFolderId", kbFolderId);

			var _command = new JsonObject()
			{
				{ "/kb.kbfolder/fetch-first-child-kb-folder", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> FetchKbFolderByUrlTitleAsync(long groupId, long parentKbFolderId, string urlTitle)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("parentKbFolderId", parentKbFolderId);
			_parameters.Add("urlTitle", urlTitle);

			var _command = new JsonObject()
			{
				{ "/kb.kbfolder/fetch-kb-folder-by-url-title", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> GetKbFolderByUrlTitleAsync(long groupId, long parentKbFolderId, string urlTitle)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("parentKbFolderId", parentKbFolderId);
			_parameters.Add("urlTitle", urlTitle);

			var _command = new JsonObject()
			{
				{ "/kb.kbfolder/get-kb-folder-by-url-title", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetKbFoldersAndKbArticlesAsync(long groupId, long parentResourcePrimKey, int status, int start, int end, JsonObjectWrapper orderByComparator)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("parentResourcePrimKey", parentResourcePrimKey);
			_parameters.Add("status", status);
			_parameters.Add("start", start);
			_parameters.Add("end", end);
			this.MangleWrapper(_parameters, "orderByComparator", "$languageUtil.getJSONWrapperClassName($parameter.type)", orderByComparator);

			var _command = new JsonObject()
			{
				{ "/kb.kbfolder/get-kb-folders-and-kb-articles", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<long> GetKbFoldersAndKbArticlesCountAsync(long groupId, long parentResourcePrimKey, int status)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("parentResourcePrimKey", parentResourcePrimKey);
			_parameters.Add("status", status);

			var _command = new JsonObject()
			{
				{ "/kb.kbfolder/get-kb-folders-and-kb-articles-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task MoveKbFolderAsync(long kbFolderId, long parentKBFolderId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("kbFolderId", kbFolderId);
			_parameters.Add("parentKBFolderId", parentKBFolderId);

			var _command = new JsonObject()
			{
				{ "/kb.kbfolder/move-kb-folder", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}

		public async Task<dynamic> GetKbFolderAsync(long kbFolderId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("kbFolderId", kbFolderId);

			var _command = new JsonObject()
			{
				{ "/kb.kbfolder/get-kb-folder", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}
	}
}

File: android/services-v73/src/main/java/com/liferay/mobile/android/v73/kbfolder/KbfolderService.java
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.mobile.android.v73.kbfolder;

import com.liferay.mobile.android.service.BaseService;
import com.liferay.mobile.android.service.JSONObjectWrapper;
import com.liferay.mobile.android.service.Session;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * @author Bruno Farache
 */
public class KbfolderService extends BaseService {

	public KbfolderService(Session session) {
		super(session);
	}

	public JSONObject addKbFolder(long groupId, long parentResourceClassNameId, long parentResourcePrimKey, String name, String description, JSONObjectWrapper serviceContext) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("description", checkNull(description));
			_params.put("groupId", groupId);
			_params.put("name", checkNull(name));
			_params.put("parentResourceClassNameId", parentResourceClassNameId);
			_params.put("parentResourcePrimKey", parentResourcePrimKey);
			mangleWrapper(_params, "serviceContext", "com.liferay.portal.kernel.service.ServiceContext", serviceContext);

			_command.put("/kb.kbfolder/add-kb-folder", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject deleteKbFolder(long kbFolderId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("kbFolderId", kbFolderId);

			_command.put("/kb.kbfolder/delete-kb-folder", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject fetchFirstChildKbFolder(long groupId, long kbFolderId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("groupId", groupId);
			_params.put("kbFolderId", kbFolderId);

			_command.put("/kb.kbfolder/fetch-first-child-kb-folder", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject fetchFirstChildKbFolder(long groupId, long kbFolderId, JSONObjectWrapper orderByComparator) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("groupId", groupId);
			_params.put("kbFolderId", kbFolderId);
			mangleWrapper(_params, "orderByComparator", "com.liferay.portal.kernel.util.OrderByComparator<com.liferay.knowledge.base.model.KBFolder>", orderByComparator);

			_command.put("/kb.kbfolder/fetch-first-child-kb-folder", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject fetchKbFolder(long kbFolderId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("kbFolderId", kbFolderId);

			_command.put("/kb.kbfolder/fetch-kb-folder", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject fetchKbFolderByUrlTitle(long groupId, long parentKbFolderId, String urlTitle) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("groupId", groupId);
			_params.put("parentKbFolderId", parentKbFolderId);
			_params.put("urlTitle", checkNull(urlTitle));

			_command.put("/kb.kbfolder/fetch-kb-folder-by-url-title", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject getKbFolder(long kbFolderId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("kbFolderId", kbFolderId);

			_command.put("/kb.kbfolder/get-kb-folder", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject getKbFolderByUrlTitle(long groupId, long parentKbFolderId, String urlTitle) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("groupId", groupId);
			_params.put("parentKbFolderId", parentKbFolderId);
			_params.put("urlTitle", checkNull(urlTitle));

			_command.put("/kb.kbfolder/get-kb-folder-by-url-title", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONArray getKbFolders(long groupId, long parentKBFolderId, int start, int end) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("end", end);
			_params.put("groupId", groupId);
			_params.put("parentKBFolderId", parentKBFolderId);
			_params.put("start", start);

			_command.put("/kb.kbfolder/get-kb-folders", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONArray(0);
	}

	public JSONArray getKbFoldersAndKbArticles(long groupId, long parentResourcePrimKey, int status, int start, int end, JSONObjectWrapper orderByComparator) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("end", end);
			_params.put("groupId", groupId);
			_params.put("parentResourcePrimKey", parentResourcePrimKey);
			_params.put("start", start);
			_params.put("status", status);
			mangleWrapper(_params, "orderByComparator", "com.liferay.portal.kernel.util.OrderByComparator", orderByComparator);

			_command.put("/kb.kbfolder/get-kb-folders-and-kb-articles", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONArray(0);
	}

	public Integer getKbFoldersAndKbArticlesCount(long groupId, long parentResourcePrimKey, int status) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("groupId", groupId);
			_params.put("parentResourcePrimKey", parentResourcePrimKey);
			_params.put("status", status);

			_command.put("/kb.kbfolder/get-kb-folders-and-kb-articles-count", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getInt(0);
	}

	public Integer getKbFoldersCount(long groupId, long parentKBFolderId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("groupId", groupId);
			_params.put("parentKBFolderId", parentKBFolderId);

			_command.put("/kb.kbfolder/get-kb-folders-count", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getInt(0);
	}

	public void moveKbFolder(long kbFolderId, long parentKBFolderId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("kbFolderId", kbFolderId);
			_params.put("parentKBFolderId", parentKBFolderId);

			_command.put("/kb.kbfolder/move-kb-folder", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		session.invoke(_command);
	}

	public JSONObject updateKbFolder(long parentResourceClassNameId, long parentResourcePrimKey, long kbFolderId, String name, String description, JSONObjectWrapper serviceContext) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("description", checkNull(description));
			_params.put("kbFolderId", kbFolderId);
			_params.put("name", checkNull(name));
			_params.put("parentResourceClassNameId", parentResourceClassNameId);
			_params.put("parentResourcePrimKey", parentResourcePrimKey);
			mangleWrapper(_params, "serviceContext", "com.liferay.portal.kernel.service.ServiceContext", serviceContext);

			_command.put("/kb.kbfolder/update-kb-folder", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

}